src/darwin/Framework/CHIP/MTRDevice_Internal.h

```diff
@@ -117,6 +117,9 @@ MTR_TESTABLE
 
 - (void)setStorageBehaviorConfiguration:(MTRDeviceStorageBehaviorConfiguration *)storageBehaviorConfiguration;
 
+// Returns whether this MTRDevice uses Thread for communication
+- (BOOL)deviceUsesThread;
```

Comment:
We should probably have a public or private bitmask for what type of transports it supports...

Reply:
(More a note, I might take a stab at this)

Reply:
Right, this is currently just used internally to signal when to pool these attempts. I think things can be named in a more generic way..... Naming is a hard problem and we definitely need to give it more thought.

Reply:
It's fine for now :)